rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -40,51 +40,68 @@ bool database_exists(const std::string & uri)
   std::ifstream database(uri);
   return database.good();
 }
+
+std::string to_string(rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  switch (io_flag) {
+    case rosbag2_storage::storage_interfaces::IOFlag::APPEND:
+      return "APPEND";
+    case rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY:
+      return "READ_ONLY";
+    case rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE:
+      return "READ_WRITE";
+    default:
+      return "UNKNOWN";
+  }
+}
+
+bool is_read_write(const rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  return io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE;
 }
 
+constexpr const auto FILE_EXTENSION = ".db3";
+}  // namespace
+
 namespace rosbag2_storage_plugins
 {
 
 void SqliteStorage::open(
   const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag)
 {
-  auto metadata = is_read_only(io_flag) ?
-    load_metadata(uri) :
-    std::unique_ptr<rosbag2_storage::BagMetadata>();
-
-  if (metadata) {
-    if (metadata->relative_file_paths.empty()) {
-      throw std::runtime_error(
-              "Failed to read from bag '" + uri + "': Missing database file path in metadata");
-    }
-
-    database_name_ = metadata->relative_file_paths[0];
-  } else {
-    if (is_read_only(io_flag)) {
-      throw std::runtime_error("Failed to read from bag '" + uri + "': No metadata found.");
-    }
-
-    database_name_ = rosbag2_storage::FilesystemHelper::get_folder_name(uri) + ".db3";
+  if (is_read_write(io_flag)) {
+    relative_path_ = uri + FILE_EXTENSION;
```

Comment:
what happens if that file already exists?

Reply:
imo: it should throw an exception.

I looked into how its handled in `SqliteWrapper` and `APPEND` and `READ_WRITE` are handled the same.

Unless you can provide an alternative, I'm going to rewrite this section to `throw std::runtime_error` when `is_read_write(io_flag) && database_exists(relative_path_)`

Reply:
@Karsten1987 I can push the changes if you're fine with it throwing when file exists and `READ_ONLY`.

Reply:
sounds good to me.